src/frontend/query_json.ml

```diff
@@ -249,6 +241,19 @@ let json_of_error (error : Location.error) =
     Location.print_main Format.str_formatter error;
     String.trim (Format.flush_str_formatter ())
   in
+  let typ =
+    match error.source with
+    | Location.Lexer   -> "lexer"
+    | Location.Parser  -> "parser"
+    | Location.Typer   -> "typer"
+    | Location.Warning ->
+      if String.is_prefixed ~by:"Error" msg then
+        "typer" (* Handle warn-error (since 4.08) *)
```

Comment:
I tried doing this earlier (in mtyper/msupport), but couldn't make it work across multiple versions of OCaml.